### PR TITLE
Fixes #53 Fixes #44 got headers issues in order

### LINKED ISSUE #44: Supporting non-json responses
I have use cases where the data that I'm requesting is not `application/json`, but is `text/html` or `application/xml`. I was planning on parsing this incoming data in `ampersand-model`'s `parse` function, and setting the model's properties there.

However, the current logic of `ampersand-sync` prevents that from happening. See the code from lines 117 of `ampersand-sync.js`:

````javascript 
if (body && typeof body === 'string') {
    try {
        body = JSON.parse(body);
    } catch (err) {
        if (options.error) return options.error(resp, 'error', err.message);  
    }
}
if (options.success) return options.success(body, 'success', resp);
````

Here, if the body is a string (true of JSON, HTML, XML…), the code will try and parse it as JSON, which  will fail for non-json, and result in `options.error` being called, thus never making it to the model's parse function.

Is there a solution which doesn't involve calling `options.error` when the data can't be parsed as JSON? It could be as simple as not doing anything in the `catch` block.

### LINKED ISSUE #53: Headers can be duplicated, sync treats them as case-sensitive
https://github.com/AmpersandJS/ampersand-sync/blob/master/core.js#L82 merges user-defined headers with defaults. If case differs, a header will be set twice. 

Also, both `xhr` and `request` always return lowercase headers

### PR DESCRIPTION
**Pull request by naugtur**
@latentflip @HenrikJoreteg @dhritzkiv Please take a look if you have a minute.

### CODE REVIEW

**Comment by pgilad** [comment]
Looks good. +1

**Comment by dhritzkiv** [comment]
Ditto; that's pretty much how I'd have done the JSON vs. not-JSON response handling

**Comment by naugtur** [comment]
My logic was: default accept is the one that gets default handling. 
The behavior we want is to parse json responses and return everything else as text. It wouldn't be a good "developer experience" if sync parsed some response types based on a potentially growing list. I would consider adding such types to the list a breaking change every time. I hear people don't like breaking changes in sync.

I used `DEFAULT_ACCEPT` to prevent "magic string" occurring twice, but you are right that we should have a better check for json type. I'm not sure if that's what you had in mind, but the server could return `application/json; charset=utf-8`and we need to match that.

**Comment by latentflip** [comment]
Aside from your addendum about charset being a potential issue (on which I trust your judgement more than mine), this lgtm

**Comment by naugtur** [comment]
I might be overdoing this, but I analyzed multiple combinations and relying on a response header would be a very breaking change for a potentially big number of users.

The decision here is surprisingly hard. Some facts first:
1. Setting a default `accept` header disables cross-domain requests for IE8-9 (IE9 still officially supported), luckily if `emulateJSON` is not used, sync will put body in xhr's `json` option, which sets accept header as needed. 
2. Remember: xhr will try parsing JSON if `json` option was used for sending it, so we might sometimes just get a readily parsed object
3. Our users relied on ampersand always parsing JSON and we want to give some users an option to opt-out
4. Responses with no content-type header used to be treated as JSON and changing that would be a breaking change
5. Responses with bogus content-types used to be treated as JSON, so same here.

Based on all that:
- no default accept header
- try parsing string as JSON unless `accept` header was explicitly set by the user and `headers['accept'].indexOf('application/json')!==0` 

It's as close to the original behavior as possible, and it brings back the option to opt-out of parsing.

If someone wants to veto and force adding a check for response content-type do it quick.

**Comment by dhritzkiv** [comment]
Could it not be argued that this breaking change is excused by an increase in major version number of this module and any modules that depend on it (ampersand-model, ampersand-rest-collection)? Users have to explicitly upgrade to the latest major version, at which point they would consult the changelog/documentation before upgrading or upon noticing the breaking behaviour. I would vote to go with whatever is the most "right", even if it breaks things (things that can be fixed by being more explicit with the request).

Ultimately, you know best as there are lots of intricacies which I can't fully comprehend.

Note: any of the proposed solutions should work for me as I'm always setting `Accept` and `Content-Type` request headers as well as `Content-Type` response headers. So, as long as ampersand-sync doesn't error by trying to parse non-JSON content as JSON, it's good.

**Comment by naugtur** [comment]
I just spent 2h getting everything to keep working in old IEs. I'm not thinking straight anymore. 
Will be back...

File: test/unit.js [hunk 1]
```diff
@@ -212,13 +206,12 @@ test('should call provided error callback on error.', function (t) {
...
 
 test('should call provided error callback is bad JSON error.', function (t) {
-    t.plan(3);
+    t.plan(2);
 
     var xhr = sync('read', modelStub(), {
         error: function (resp, type, error) {
             t.deepEqual(resp, {}, 'should be passed through response');
             t.equal(type, 'error', 'is string \'error\' as per jquery');
-            t.ok(error == 'Unable to parse JSON string' || error == 'Unexpected end of input', 'should be json parse message');
```
Comment (naugtur) [inline test/unit.js]: I found another new message for this error recently, no point checking that anymore, it'd fail sooner or later

**Comment by naugtur** [comment]
Last blocker here: I'm putting `isEmpty` check in xhr and releasing it together with some minor fixes. Will test in IE tomorrow and publish.
Once it's done, I'm merging this PR and everything else is just simple chores. ;)

**Comment by HenrikJoreteg** [comment]
@naugtur sounds great, thanks for your hard work on this!

**Comment by naugtur** [comment]
I made the change in XHR, will give everyone the weekend to comment on PR there, while I'm on a bit of vacation. I want to get new changes to xhr2 published and this PR finalized on Monday.

**Pull request merged**